opnode/test/system_test.go

```diff
@@ -641,3 +645,196 @@ func TestL1InfoContract(t *testing.T) {
+	defer cancel()
+	startBalance, err = l2Seq.BalanceAt(ctx, fromAddr, nil)
+	require.Nil(t, err)
+
+	// Intiate Withdrawal
+	withdrawAmount := big.NewInt(500_000_000_000)
+	l2opts, err := bind.NewKeyedTransactorWithChainID(ethPrivKey, cfg.L2ChainID)
+	require.Nil(t, err)
+	l2opts.Value = withdrawAmount
+	tx, err = l2withdrawer.InitiateWithdrawal(l2opts, fromAddr, big.NewInt(21000), nil)
+	require.Nil(t, err, "sending initiate withdraw tx")
+
+	receipt, err = waitForTransaction(tx.Hash(), l2Seq, 3*time.Duration(cfg.L1BlockTime)*time.Second)
+	require.Nil(t, err, "withdrawal initiated on L2 sequencer")
+	require.Equal(t, receipt.Status, types.ReceiptStatusSuccessful, "transaction failed")
+
+	// Verify L2 balance after withdrawal
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	header, err := l2Seq.HeaderByNumber(ctx, receipt.BlockNumber)
+	require.Nil(t, err)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	endBalance, err = l2Seq.BalanceAt(ctx, fromAddr, nil)
+	require.Nil(t, err)
+
+	// Take fee into account
+	diff = new(big.Int).Sub(startBalance, endBalance)
+	fees := calcGasFees(receipt.GasUsed, tx.GasTipCap(), tx.GasFeeCap(), header.BaseFee)
+	diff = diff.Sub(diff, fees)
+	require.Equal(t, withdrawAmount, diff)
+
+	// Take start balance on L1
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	startBalance, err = l1Client.BalanceAt(ctx, fromAddr, nil)
+	require.Nil(t, err)
+
+	// Wait for finalization and then create the Finalized Withdrawal Transaction
+	l2OutputOracle, err := l2oo.NewL2OutputOracleCaller(sys.L2OOContractAddr, l1Client)
+	require.Nil(t, err)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 10*time.Duration(cfg.L1BlockTime)*time.Second)
+	defer cancel()
+	timestamp, err := withdrawals.WaitForFinalizationPeriod(ctx, l1Client, sys.DepositContractAddr, header.Time)
+	require.Nil(t, err)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	blockNumber, err := l2OutputOracle.ComputeL2BlockNumber(&bind.CallOpts{Context: ctx}, new(big.Int).SetUint64(timestamp))
+	require.Nil(t, err)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	header, err = l2Seq.HeaderByNumber(ctx, blockNumber)
+	require.Nil(t, err)
+
+	rpc, err := rpc.Dial(cfg.Nodes["sequencer"].L2NodeAddr)
+	require.Nil(t, err)
+	l2client := withdrawals.NewClient(rpc)
+
+	// Now create withdrawal
+	params, err := withdrawals.FinalizeWithdrawalParameters(context.Background(), l2client, tx.Hash(), header)
+	require.Nil(t, err)
+
+	portal, err := deposit.NewOptimismPortal(sys.DepositContractAddr, l1Client)
+	require.Nil(t, err)
+
+	opts.Value = nil
+	tx, err = portal.FinalizeWithdrawalTransaction(
+		opts,
+		params.Nonce,
+		params.Sender,
+		params.Target,
+		params.Value,
+		params.GasLimit,
+		params.Data,
```

Comment:
Might want to test a `params.Data` that's not empty, but integration tests can pick that up too.

Reply:
Good call - that'd be part of doing generic contract calls (rather than simple transfer) via the withdrawal mechanism